examples/crd_derive_schema.rs

```diff
@@ -0,0 +1,146 @@
+}
+
+fn main() {
+    // Show the generated CRD
+    println!("{}", serde_yaml::to_string(&Foo::crd()).unwrap());
+}
+
+#[test]
+fn verify_crd_schema() {
+    let crd = Foo::crd();
+    let output = serde_json::json!({
+        "apiVersion": "apiextensions.k8s.io/v1",
+        "kind": "CustomResourceDefinition",
+        "metadata": {
+            "name": "foos.clux.dev"
+        },
+        "spec": {
+            "group": "clux.dev",
+            "names": {
+                "kind": "Foo",
+                "plural": "foos",
+                "shortNames": [],
+                "singular": "foo"
+            },
+            "scope": "Namespaced",
+            "versions": [
+                {
+                    "name": "v1",
+                    "served": true,
+                    "storage": true,
+                    "additionalPrinterColumns": [],
+                    "schema": {
+                        "openAPIV3Schema": {
+                            "description": "Auto-generated derived type for FooSpec via `CustomResource`",
+                            "properties": {
+                                "spec": {
+                                    "properties": {
+                                      "nullable": {
+                                          "nullable": true,
+                                          "type": "string"
+                                      },
+                                      "nullable_with_default": {
+                                          "default": "default_nullable",
+                                          "nullable": true,
+                                          "type": "string"
+                                      },
+                                      "required_field": {
+                                          "type": "string"
+                                      },
+                                      "with_default": {
+                                          "default": "default_value",
+                                          "type": "string"
+                                      },
+                                    },
+                                    "required": [
+                                        "required_field"
+                                    ],
+                                    "type": "object"
+                                }
+                            },
+                            "required": [
+                                "spec"
+                            ],
+                            "title": "Foo",
+                            "type": "object"
+                        }
+                    },
+                    "subresources": {},
+                }
+            ]
+        }
+    });
+    let expected = serde_json::from_value(output).unwrap();
+    // println!("{}", serde_yaml::to_string(&crd).unwrap());
+    // println!("{}", serde_yaml::to_string(&expected).unwrap());
+    assert_eq!(crd, expected);
```

Comment:
Question: This only tests deserialization. Is there a reason why serialization can't be tested as well ?

Reply:
Like
```rust
assert_eq!(crd, expected);
assert_eq!(serde_yaml::to_string(&crd).unwrap(), serde_yaml::to_string(&expected).unwrap());
```
?
Can you elaborate? I'm not sure what the point is. It can be added if needed.

Reply:
> This only tests deserialization.

This test case is testing the generated schema (in `Foo::crd()`) matches the expected output.